src/lib/color-picker/color-picker-selector.component.ts

```diff
@@ -427,179 +604,235 @@ export class MccColorPickerSelectorComponent
       this._alphaContext.fillStyle = this._alphaContext.createPattern(background, 'repeat');
       this._alphaContext.fillRect(0, 0, this._alpha.nativeElement.width, this._alpha.nativeElement.height);
       // if rgb is set and have at least 3 values set, create linear gradient
-      if (Object.keys(rgb).filter(k => Number.isInteger(rgb[k])).length >= 3) {
-        const alphaGrd2 = this._alphaContext.createLinearGradient(0, 0, 0, this._bc.nativeElement.height);
-        alphaGrd2.addColorStop(0, `rgba(${rgb['R']}, ${rgb['G']}, ${rgb['B']})`);
-        alphaGrd2.addColorStop(1, `rgba(${rgb['R']}, ${rgb['G']}, ${rgb['B']}, 0)`);
 
-        this._alphaContext.fillStyle = alphaGrd2;
-        this._alphaContext.fillRect(0, 0, this._alpha.nativeElement.width, this._alpha.nativeElement.height);
-      }
+      const alphaGrd2 = this._alphaContext.createLinearGradient(0, 0, 0, this._bc.nativeElement.height);
+      alphaGrd2.addColorStop(0, `rgba(${color.toRgb().r}, ${color.toRgb().g}, ${color.toRgb().b}, 1)`);
+      alphaGrd2.addColorStop(1, `rgba(${color.toRgb().r}, ${color.toRgb().g}, ${color.toRgb().b}, 0)`);
+
+      this._alphaContext.fillStyle = alphaGrd2;
+      this._alphaContext.fillRect(0, 0, this._alpha.nativeElement.width, this._alpha.nativeElement.height);
     };
   }
 
   /**
-   * Generate colors based on the RGBA color
+   * Generate color selector block based on the RGB color
    */
-  private _fillGradient(): void {
-    this._blockContext.fillStyle = this._rgbaColor;
+  private _drawBlockSelector(hueColor: Instance) {
+    this._blockContext.fillStyle = `hsl(${hueColor.toHsl().h}, 100%, 50%)`;
     this._blockContext.fillRect(0, 0, this._bc.nativeElement.width, this._bc.nativeElement.height);
 
-    const grdWhite = this._stripContext.createLinearGradient(0, 0, this._bc.nativeElement.width, 0);
+    // TODO: hsl selector #145
+    // const grdLeft = this._hueSelectorContext.createLinearGradient(
+    //   0,
+    //   0,
+    //   this._bc.nativeElement.width,
+    //   0
+    // );
+    // grdLeft.addColorStop(0, 'hsl(0, 0%, 50%)');
+    // grdLeft.addColorStop(1, 'hsla(0, 0%, 50%, 0)');
+    // this._blockContext.fillStyle = grdLeft;
+    // this._blockContext.fillRect(0, 0, this._bc.nativeElement.width, this._bc.nativeElement.height);
+    //
+    // const grdTop = this._hueSelectorContext.createLinearGradient(0, 0, 0, this._bc.nativeElement.height);
+    // grdTop.addColorStop(0, 'hsl(0, 0%, 100%)');
+    // grdTop.addColorStop(0.5, 'hsla(0, 0%, 100%, 0)');
+    // grdTop.addColorStop(0.5, 'hsla(0, 0%, 0%, 0)');
+    // grdTop.addColorStop(1, 'hsl(0, 0%, 0%)');
+    // this._blockContext.fillStyle = grdTop;
+    // this._blockContext.fillRect(0, 0, this._bc.nativeElement.width, this._bc.nativeElement.height);
```

Comment:
Please remove this commented code from this PR.

Reply:
I pushed your requested changes